Fail
file path is :
projects/deltachat-core/rust/src/message.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/RustRepoTrans/projects/deltachat-core/rust)
error[E0252]: the name `Result` is defined multiple times
  --> src/message.rs:13:48
   |
5  | use anyhow::Result;
   |     -------------- previous import of the type `Result` here
...
13 | use anyhow::{ensure, format_err, Context as _, Result};
   |                                                ^^^^^^ `Result` reimported here
   |
   = note: `Result` must be defined only once in the type namespace of this module

error[E0255]: the name `Message` is defined multiple times
   --> src/message.rs:427:1
    |
6   | use crate::message::Message;
    |     ----------------------- previous import of the type `Message` here
...
427 | pub struct Message {
    | ^^^^^^^^^^^^^^^^^^ `Message` redefined here
    |
    = note: `Message` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
    |
6   | use crate::message::Message as OtherMessage;
    |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0252]: the name `EventType` is defined multiple times
  --> src/message.rs:31:5
   |
8  | use crate::EventType;
   |     ---------------- previous import of the type `EventType` here
...
31 | use crate::events::EventType;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ `EventType` reimported here
   |
   = note: `EventType` must be defined only once in the type namespace of this module

error[E0432]: unresolved imports `crate::MessageState`, `crate::MessageState`
 --> src/message.rs:7:12
  |
7 | use crate::MessageState::{self, *};
  |            ^^^^^^^^^^^^   ^^^^ no `MessageState` in the root
  |            |
  |            could not find `MessageState` in the crate root

error[E0425]: cannot find value `OutFailed` in this scope
    --> src/message.rs:1759:21
     |
1759 |         msg.state = OutFailed;
     |                     ^^^^^^^^^ not found in this scope
     |
help: consider importing this unit variant
     |
4    + use crate::message::MessageState::OutFailed;
     |

error[E0425]: cannot find value `DC_PARAM_ERROR` in this scope
    --> src/message.rs:1763:33
     |
1763 |         dc_param_set(msg.param, DC_PARAM_ERROR, error).await?;
     |                                 ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `SQLITE_STATIC` in this scope
    --> src/message.rs:1769:63
     |
1769 |     sqlite3_bind_text(stmt, 2, msg.param.packed.as_str(), -1, SQLITE_STATIC).await?;
     |                                                               ^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_EVENT_MSG_FAILED` in this scope
    --> src/message.rs:1773:25
     |
1773 |     context.cb(context, DC_EVENT_MSG_FAILED, msg.chat_id, msg.id).await?;
     |                         ^^^^^^^^^^^^^^^^^^^ not found in this scope

warning: unused import: `std::sync::Arc`
 --> src/message.rs:4:5
  |
4 | use std::sync::Arc;
  |     ^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:6:5
  |
6 |     unused,
  |     ^^^^^^
  = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

warning: unused import: `Result`
  --> src/message.rs:13:48
   |
13 | use anyhow::{ensure, format_err, Context as _, Result};
   |                                                ^^^^^^

warning: unused import: `crate::events::EventType`
  --> src/message.rs:31:5
   |
31 | use crate::events::EventType;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved imports `crate::MessageState`, `crate::MessageState`
    --> src/message.rs:7:12
     |
7    | use crate::MessageState::{self, *};
     |            ^^^^^^^^^^^^   ^^^^ no `MessageState` in the root
     |            |
     |            could not find `MessageState` in the crate root
     |
note: enum `crate::test_utils::tests::MessageState` exists but is inaccessible
    --> src/message.rs:1272:1
     |
1272 | pub enum MessageState {
     | ^^^^^^^^^^^^^^^^^^^^^ not accessible

error[E0425]: cannot find value `OutFailed` in this scope
    --> src/message.rs:1759:21
     |
1759 |         msg.state = OutFailed;
     |                     ^^^^^^^^^ not found in this scope
     |
note: unit variant `crate::test_utils::tests::MessageState::OutFailed` exists but is inaccessible
    --> src/message.rs:1305:5
     |
1305 |     OutFailed = 24,
     |     ^^^^^^^^^^^^^^ not accessible

error[E0425]: cannot find function `dc_msg_new_untyped` in this scope
    --> src/message.rs:1753:15
     |
1753 |     let msg = dc_msg_new_untyped(context).await?;
     |               ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_msg_load_from_db` in this scope
    --> src/message.rs:1754:9
     |
1754 |     if !dc_msg_load_from_db(msg, context, msg.id).await? {
     |         ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0308]: mismatched types
    --> src/message.rs:1762:12
     |
1762 |     if let Some(error) = error {
     |            ^^^^^^^^^^^   ----- this expression has type `&str`
     |            |
     |            expected `str`, found `Option<_>`
     |
     = note: expected type `str`
                found enum `std::option::Option<_>`

error[E0425]: cannot find function `dc_param_set` in this scope
    --> src/message.rs:1763:9
     |
1763 |         dc_param_set(msg.param, DC_PARAM_ERROR, error).await?;
     |         ^^^^^^^^^^^^ not found in this scope

error[E0277]: `()` is not a future
    --> src/message.rs:1764:37
     |
1764 |         warn!(context, "{}", error).await?;
     |                                    -^^^^^
     |                                    ||
     |                                    |`()` is not a future
     |                                    help: remove the `.await`
     |
     = help: the trait `futures::Future` is not implemented for `()`, which is required by `(): std::future::IntoFuture`
     = note: () must be a future or must implement `IntoFuture` to be awaited
     = note: required for `()` to implement `std::future::IntoFuture`

error[E0425]: cannot find function `dc_sqlite3_prepare` in this scope
    --> src/message.rs:1767:16
     |
1767 |     let stmt = dc_sqlite3_prepare(context.sql, "UPDATE msgs SET state=?, param=? WHERE id=?;").await?;
     |                ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `sqlite3_bind_int` in this scope
    --> src/message.rs:1768:5
     |
1768 |     sqlite3_bind_int(stmt, 1, msg.state as i32).await?;
     |     ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `sqlite3_bind_text` in this scope
    --> src/message.rs:1769:5
     |
1769 |     sqlite3_bind_text(stmt, 2, msg.param.packed.as_str(), -1, SQLITE_STATIC).await?;
     |     ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `sqlite3_bind_int` in this scope
    --> src/message.rs:1770:5
     |
1770 |     sqlite3_bind_int(stmt, 3, msg.id).await?;
     |     ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `sqlite3_step` in this scope
    --> src/message.rs:1771:5
     |
1771 |     sqlite3_step(stmt).await?;
     |     ^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `cb` found for reference `&context::Context` in the current scope
    --> src/message.rs:1773:13
     |
1773 |     context.cb(context, DC_EVENT_MSG_FAILED, msg.chat_id, msg.id).await?;
     |             ^^ method not found in `&Context`

error[E0425]: cannot find function `sqlite3_finalize` in this scope
    --> src/message.rs:1775:5
     |
1775 |     sqlite3_finalize(stmt).await?;
     |     ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `dc_msg_unref` in this scope
    --> src/message.rs:1776:5
     |
1776 |     dc_msg_unref(msg).await?;
     |     ^^^^^^^^^^^^ not found in this scope

Some errors have detailed explanations: E0252, E0255, E0277, E0308, E0425, E0432, E0599.
For more information about an error, try `rustc --explain E0252`.
warning: `deltachat` (lib) generated 3 warnings
error: could not compile `deltachat` (lib) due to 21 previous errors; 3 warnings emitted
warning: build failed, waiting for other jobs to finish...
warning: `deltachat` (lib test) generated 3 warnings (3 duplicates)
error: could not compile `deltachat` (lib test) due to 21 previous errors; 3 warnings emitted
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

